drivers/display/display_stm32_ltdc.c

```diff
@@ -169,6 +170,15 @@ static void stm32_ltdc_get_capabilities(const struct device *dev,
 
 	capabilities->current_pixel_format = data->current_pixel_format;
 	capabilities->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+
+#if CONFIG_ILI9341
+	const struct device *ili9341 = DEVICE_DT_GET(DT_INST(0, ilitek_ili9341));
+
+	/* Needed for display to function correctly on stm32f429i_disc1 */
+	if (device_is_ready(ili9341)) {
+		display_blanking_off(ili9341);
+	}
+#endif /* CONFIG_ILI9341 */
```

Comment:
no need to hard code a specific device-to-device dependency, do this instead:

- add an optional `phandle` property for the secondary device in `st,stm32-ltdc.yaml`
- use that to point at `ili9341` from the `&ltdc` node
- in this file use the property to populate a `struct device` pointer (or NULL) in the `display_stm32_ltdc_config` structure
- in `stm32_ltdc_display_api` implement the `.blanking_on` and `.blanking_off` function, there you can check if you have a device in that pointer, and if you do call `display_blanking_on` and `display_blanking_off` in that device (which will result in calling the right function at the right time once the application tries to unblank the ltdc controller)